framework/src/userobjects/TimedSubdomainModifier.C

```diff
@@ -0,0 +1,298 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TimedSubdomainModifier.h"
+#include "TimedElementSubdomainModifier.h"
+#include "DelimitedFileReaderOfString.h"
+#include "MooseMesh.h"
+
+registerMooseObject("MooseApp", TimedSubdomainModifier);
+
+InputParameters
+TimedSubdomainModifier::validParams()
+{
+  InputParameters params = TimedElementSubdomainModifier::validParams();
+
+  // parameters for direct input (additionally to 'times')
+  params.addParam<std::vector<std::string>>("blocks_from", "Names or ids of the 'old' block.");
+  params.addParam<std::vector<std::string>>("blocks_to", "Names or ids of the 'new' block.");
+  params.addParamNamesToGroup("times blocks_from blocks_to", "Direct data input.");
+
+  // parameters for file-based data supply
+  params.addParam<FileName>("data_file", "File holding CSV data");
+  params.addParam<bool>("header",
+                        "Indicates whether the file contains a header with the column names");
+  params.addParam<std::string>("delimiter", ",", "Delimiter used to parse the file");
+  params.addParam<std::string>("comment", "#", "Comment character used to parse the file");
```

Comment:
you should be able to get these parameters from another object? as in
params += DelimitedFileReader::validParams() ?

Reply:
'params += DelimitedFileReader::validParams()' for sure looks much cleaner. But DelimitedFileReader exposes 'format_flag' current not supported by TimedSubdomainModifier. How do I pick the supported parameters?

Reply:
you suppress parameters with params.suppress....()

Reply:
Here I got a small problem: 'validParams' is not a member of 'DelimitedFileReader' (I presume this is due to the fact its in 'utils').

Reply:
ah good point. I ll see to it later